edb/edgeql/compiler/expr.py

```diff
@@ -523,11 +523,20 @@ def compile_TypeCast(
                 # JSON wants type shapes and acts as an output sink.
                 subctx.expr_exposed = True
                 subctx.inhibit_implicit_limit = True
+
+            if (
+                isinstance(expr.type, qlast.TypeName) and
+                isinstance(expr.type.maintype, qlast.ObjectRef) and
+                expr.type.maintype.name == 'json'
```

Comment:
Is there a better way to check if the target type is `std::json`?

Reply:
The `if` check above (`if target_stype.contains_json(subctx.env.schema):`) is doing exactly that.

Reply:
changed